test/contrib/emscripten/conftest.py

```diff
@@ -90,44 +141,81 @@ def _install_coverage(self: Any) -> None:
 
 
 class ServerRunnerInfo:
-    def __init__(self, host: str, port: int, selenium: Any) -> None:
+    def __init__(self, host: str, port: int, selenium: Any, dist_dir: Path) -> None:
         self.host = host
         self.port = port
         self.selenium = selenium
+        self.dist_dir = dist_dir
 
-    def run_webworker(self, code: str) -> Any:
+    def run_webworker(self, code: str, *, has_jspi: bool = True) -> Any:
         if isinstance(code, str) and code.startswith("\n"):
             # we have a multiline string, fix indentation
             code = textwrap.dedent(code)
-            # add coverage collection to this code
+        # make sure any import of urllib comes from our package
+        code = (
+            textwrap.dedent(
+                f"""
+                import pyodide_js as _pjs
+                await _pjs.loadPackage('https://{self.host}:{self.port}/dist/urllib3.whl')
+                """
+            )
+            + code
+        )
+
+        if has_jspi is False:
+            # disable jspi in this code
```

Comment:
Why are we monkey patching `has_jspi()` here? Shouldn't `has_jspi()` correctly detect this? In real life if people are using Firefox or Safari or some other runtime without JSPI, they won't get this monkey patch.

Reply:
It was just to save loading different browsers and force the jspi code to be executed. I simplified the code relating to this now, so it uses the non-jspi selenium instead.